main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1601,17 +1601,20 @@ object Typer {
 
 
       case KindedAst.Expression.NewChannel(exp, elmType, loc) =>
+        val regionVar = Type.freshVar(Kind.Bool, loc, text = FallbackText("region"))
+
         for {
           (constrs, tpe, _, eff) <- visitExp(exp)
           _ <- expectTypeM(expected = Type.Int32, actual = tpe, exp.loc)
-          resultTyp <- liftM(Type.mkTuple(List(Type.mkSender(elmType, loc), Type.mkReceiver(elmType, loc)), loc))
+          resultTyp <- liftM(Type.mkTuple(List(Type.mkSender(elmType, regionVar, loc), Type.mkReceiver(elmType, regionVar, loc)), loc))
```

Comment:
Here we are allocating a new channel (and for now) that is always in the static region, hence regionVar here should really be Type.False. (This will change when you later add regions to NEW_CHANNEL)